nhrpd/nhrp_vty.c

```diff
@@ -594,29 +595,62 @@ struct info_ctx {
 	struct vty *vty;
 	afi_t afi;
 	int count;
+	struct json_object *json;
 };
 
 static void show_ip_nhrp_cache(struct nhrp_cache *c, void *pctx)
 {
 	struct info_ctx *ctx = pctx;
 	struct vty *vty = ctx->vty;
 	char buf[2][SU_ADDRSTRLEN];
+	struct json_object *json = NULL;
 
 	if (ctx->afi != family2afi(sockunion_family(&c->remote_addr)))
 		return;
 
-	if (!ctx->count) {
+	if (ctx->json)
+		json = json_object_new_object();
+	if (!ctx->count && !ctx->json) {
 		vty_out(vty, "%-8s %-8s %-24s %-24s %-6s %s\n", "Iface", "Type",
 			"Protocol", "NBMA", "Flags", "Identity");
 	}
 	ctx->count++;
 
+	sockunion2str(&c->remote_addr, buf[0], sizeof(buf[0]));
+	if (c->cur.peer)
+		sockunion2str(&c->cur.peer->vc->remote.nbma,
+			      buf[1], sizeof(buf[1]));
+	else
+		snprintf(buf[1], sizeof(buf[1]), "-");
+	if (json) {
+		json_object_string_add(json, "interface", c->ifp->name);
+		json_object_string_add(json, "type",
+				       nhrp_cache_type_str[c->cur.type]);
+		json_object_string_add(json, "protocol", buf[0]);
+		json_object_string_add(json, "nbma", buf[1]);
+		if (c->used)
+			json_object_boolean_true_add(json, "used");
+		else
+			json_object_boolean_false_add(json, "used");
+		if (c->t_timeout)
```

Comment:
me as always with pedantic changes: can we have an extra blank line between IFs? :)

Reply:
you will tell me after change.
if you really want this, just let s see how to add it in codestyle checker.